doc/model.rst

```diff
@@ -0,0 +1,683 @@
+``Model.attach_generator_files()`` function and specify the `to_configure`
+parameter with the path to the text file, `configure_params.txt`:
+
+.. code-block:: python
+
+    # Attach the file to the Model instance
+    model.attach_generator_files(to_configure="path/to/configure_params.txt")
+
+When we launch the Model, using ``Experiment.start()``, the file, `configure_inputs.txt`,
+will be in the same file path as the application and SmartSim will read and assign
+Model parameters from the text file.
+
+=====================
+ML Models and Scripts
+=====================
+--------
+Overview
+--------
+SmartSim supports sending TorchScript functions, scripts, and
+TensorFlow (TF), TensorFlow Lite (TF-lite), PyTorch (PT), or ONNX models to the database at runtime
+prior to the execution of a Model for use within the workload.
+The ``Model API`` provides a subset of helper functions that support
+these capabilities:
+
+* ``Model.add_ml_model()`` : Load a TF, TF-lite, PT, or ONNX model into the DB at runtime.
+* ``Model.add_function()`` : Launch a TorchScript function with the Model.
+* ``Model.add_script()`` : Launch a TorchScript with the Model.
+
+---------
+AI Models
+---------
+When configuring a Model, users can instruct SmartSim to load
+TensorFlow (TF), TensorFlow Lite (TF-lite), PyTorch (PT), or ONNX
+models dynamically to the database (colocated or standard). Machine Learning (ML) models added
+are loaded prior to the execution of the model. SmartSim users may
+provide the model in memory or specify the file path via the
+``Model.add_ml_model()`` Model API helper function.
+
+When specifying an ML model using ``Model.add_ml_model()``, the
+following arguments are offered:
+
+- name (str): key to store model under
+- backend (str): name of the backend (TORCH, TF, TFLITE, ONNX)
+- model (byte string, optional): A model in memory (only supported for non-colocated orchestrators)
+- model_path (file path to model): serialized model
+- device (str, optional): name of device for execution, defaults to “CPU”
+- devices_per_node (int): The number of GPU devices available on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+- first_device (int): The first GPU device to use on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+- batch_size (int, optional): batch size for execution, defaults to 0
+- min_batch_size (int, optional): minimum batch size for model execution, defaults to 0
+- min_batch_timeout (int, optional): time to wait for minimum batch size, defaults to 0
+- tag (str, optional): additional tag for model information, defaults to “”
+- inputs (list[str], optional): model inputs (TF only), defaults to None
+- outputs (list[str], optional): model outputs (TF only), defaults to None
+
+These arguments provide details to add and configure
+ML models within the model simulation.
+
+Example: Loading an In-Memory ML Model to the Model
```

Comment:
Examples here are not consistent with examples above

here:

foo
---

vs

-------
Example
-------

Reply:
yes, this is bc I wanted the example to be nested within the AI Model tab